src/runtime/HalideBuffer.h

```diff
@@ -1938,89 +1942,72 @@ class Buffer {
     // (all of different types), advance the pointers using the
     // strides.
     template<typename Ptr, typename ...Ptrs>
+    HALIDE_ALWAYS_INLINE
     static void advance_ptrs(const int *stride, Ptr *ptr, Ptrs... ptrs) {
         (*ptr) += *stride;
         advance_ptrs(stride + 1, ptrs...);
     }
 
+    HALIDE_ALWAYS_INLINE
     static void advance_ptrs(const int *) {}
 
     // Same as the above, but just increments the pointers.
     template<typename Ptr, typename ...Ptrs>
+    HALIDE_ALWAYS_INLINE
     static void increment_ptrs(Ptr *ptr, Ptrs... ptrs) {
         (*ptr)++;
         increment_ptrs(ptrs...);
     }
 
+    HALIDE_ALWAYS_INLINE
     static void increment_ptrs() {}
 
-    // Given a bunch of pointers to buffers of different types, read
-    // out their strides in the d'th dimension, and assert that their
-    // sizes match in that dimension.
-    template<typename T2, int D2, typename ...Args>
-    void extract_strides(int d, int *strides, const Buffer<T2, D2> &first, Args&&... rest) const {
-        assert(first.dimensions() == dimensions());
-        assert(first.dim(d).min() == dim(d).min() &&
-               first.dim(d).max() == dim(d).max());
-        *strides++ = first.dim(d).stride();
-        extract_strides(d, strides, std::forward<Args>(rest)...);
-    }
-
-    void extract_strides(int d, int *strides) const {}
-
-    // The template function that constructs the loop nest for for_each_value
-    template<int d, bool innermost_strides_are_one, typename Fn, typename... Ptrs>
-    static void for_each_value_helper(Fn &&f, const for_each_value_task_dim<sizeof...(Ptrs)> *t, Ptrs... ptrs) {
+    template<typename Fn, typename... Ptrs>
+    HALIDE_NEVER_INLINE
+    static void for_each_value_helper(Fn &&f, int d, bool innermost_strides_are_one,
+                                      const for_each_value_task_dim<sizeof...(Ptrs)> *t, Ptrs... ptrs) {
+        // When we hit a low dimensionality, switch from runtime
+        // recursion to template recursion.
+        asm volatile ("# Hi");
```

Comment:
Um, this absolutely needs a comment. Did you mean to leave this in?

Reply:
Ha, no I did not.

Reply:
Pretty sure this is what's breaking the windows buildbots, btw